lib/geocoder/lookups/base.rb

```diff
@@ -84,7 +84,7 @@ def query_url(query)
       #
       def cache
         if @cache.nil? and store = configuration.cache
-          @cache = Cache.new(store, configuration.cache_prefix)
+          @cache = Cache.new(store, configuration.except(:cache))
```

Comment:
I think we should be a little more careful here and only pass the relevant options. There are only two, right?

Reply:
@alexreisner I thought about passing all options just in case they're been used by different services, so if influxdb support is going to be added later -as example- it might have it's own configs which is might be different than redis options

Reply:
That is definitely true. However, I still think it's better to pass the arguments explicitly, and add new ones as new caching services require them. Otherwise we're passing options like `:use_https` to the cache service, which is potentially confusing because it sounds like a valid option (connect to Redis via HTTPS?) but certainly not what the user intended (connect to geocoding API via HTTPS). The same goes for `:logger` and `:host`...

Maybe a better solution, to allow for expansion in the future, is to add a `:cache_options` config param with a hash value instead of `:cache_expiration`. So config would look something like this:

    Geocoder.configure(
      cache: Redis.current,
      cache_options: {
        expiration: 2.days,
        prefix: 'geocoder:',
        other_option: ...
      }
    )

and the whole `cache_options` hash would be passed to the cache service. For backward compatibility, we'd need to continue supporting the `:cache_prefix` option at the root level... What do you think of this?

Reply:
@alexreisner I think you're totally right, I also agree on your suggestion, I'll implement it

Reply:
@alexreisner I think you're totally right, I also agree on your suggestion, but I think going with that suggestion will break backward compatibility, so people who were using `cache_prefix` will have a toubles, so I think we can leave `cache_prefix` outside for now, and also I'll copy it inside cache_options if not provided just as a compatibility trick, what do you think?
```ruby
Geocoder.configure(
  lookup: :google,
  api_key: ENV['GMAPS_PLAECS_KEY'].try(:freeze),
  cache: Redis.current,
  cache_prefix: "GeocoderCache:"
)```

Reply:
@alexreisner what do you thinks about https://github.com/alexreisner/geocoder/pull/1532/commits/72a6e379fcd449835004d4854f6df30589c94ace ?

Reply:
@alexreisner let me know your feedback about https://github.com/alexreisner/geocoder/pull/1532/commits/5426ff13ad7d4138a21c300c19b848dec2b52344